### ISSUE TITLE
Document how to create a targetRange.

### ISSUE THREAD

**Issue by kristoferjoseph** [facebook/draft-js#455]
In the Entities documentation there is a reference to `targetRange` but there is no documentation on what is required to create one.

https://facebook.github.io/draft-js/docs/advanced-topics-entities.html#content

```
const key = Entity.create('LINK', 'MUTABLE', {href: 'http://www.zombo.com'});
const contentStateWithLink = Modifier.applyEntity(
  contentState,
  targetRange,
  key
);
```

This has been brought up previously in #409

**Comment by maxmatthews** [comment]
I'm also struggling with this right now. I've figured out how to create an entity if text is selected with the following code:
```javascript 
var entity = Entity.create('COPYBIT', 'IMMUTABLE', {someData: data});
this.setState({
    editorState: RichUtils.toggleLink(
        this.state.editorState,
        this.state.editorState.getSelection(),
        entity
    )
});
```
but I'm struggling on how to create an entity on a fragment I've created manually.
For example, this code successfully inserts the fragment but I can't understand how to add the entity to it.
```javascript
var entity = Entity.create('COPYBIT', 'IMMUTABLE', {data: someData});

const {editorState} = this.state;
const blockMap = ContentState.createFromText(text).blockMap;
const selection = editorState.getSelection();
let newState = Draft.Modifier.replaceWithFragment(editorState.getCurrentContent(), selection, blockMap);

this.onChange(EditorState.push(editorState, newState, 'insert-fragment'));
```

**Comment by kristoferjoseph** [comment]
@maxmatthews Yup that is where I got stumped as well. Still not sure how create an entity from a programmatically created text selection. Would be great to have how to create a targetRange documented.

Use case: I want to identify user names while someone is typing and programmatically select and add styling to them.

**Comment by maxmatthews** [comment]
Here's how I got it to work. Hopefully it helps @kristoferjoseph out! 
```javascript
var copyBitEntity = Entity.create('COPYBIT', 'IMMUTABLE', {
    copyBitID: newCopyBitID
});

const {
    editorState
} = this.state;
const tempContentState = ContentState.createFromText(text);
const blockMap = tempContentState.blockMap;
const selection = editorState.getSelection();
let newState = Draft.Modifier.replaceWithFragment(editorState.getCurrentContent(), selection, blockMap);

const selectionState = new Draft.SelectionState({
    anchorKey: selection.getAnchorKey(),
    anchorOffset: selection.getStartOffset(),
    focusKey: selection.getAnchorKey(),
    focusOffset: selection.getStartOffset() + text.length
});

newState = Draft.Modifier.applyEntity(newState, selectionState, copyBitEntity);

this.onChange(EditorState.push(editorState, newState, 'insert-fragment'));
```

**Comment by kristoferjoseph** [comment]
`Draft.modifier.replaceWithFragment` using `blockMap` interesting.
What a crazy convoluted API.

**Comment by maxmatthews** [comment]
I can't tell you where I stole that code from (I think I combined two GH issues with the official docs and some help from the Slack channel) but it works and I hope it helps you. It definitely is a complicated API but I don't have any solutions on how to make it easier to use. It just has so many features that make sense in theory but are convoluted to manipulate. It's a perfect fit for my project, regardless of how hard I find it to use. 
I've found it's sometimes easier to `convertToRaw` make changes to the raw JS object then convert it back and set the editor state.
